rustfmt.toml

```diff
@@ -0,0 +1,18 @@
+# Basic
+hard_tabs = true
+max_width = 100
+use_small_heuristics = "Max"
+# Imports
+group_imports = "StdExternalCrate"
+imports_granularity = "Crate"
+reorder_imports = true
+# Consistency
+newline_style = "Unix"
+normalize_comments = true
+normalize_doc_attributes = true
+# Misc
+chain_width = 80
+spaces_around_ranges = true
+binop_separator = "Back"
+reorder_impl_items = true
```

Comment:
I think this set of changes gets us to compliance with the stype guide ( https://github.com/paritytech/substrate/blob/master/docs/STYLE_GUIDE.md ) with the exception that the 'where' block is not indented by one level. I think this is something we can get used to and we might appreciate the additional width given the size of some of our where clauses.

```suggestion
imports_granularity = "Crate"
reorder_imports = true
# Consistency
newline_style = "Unix"
normalize_comments = true
normalize_doc_attributes = true
# Misc
chain_width = 80
spaces_around_ranges = false
binop_separator = "Back"
reorder_impl_items = false
match_arm_leading_pipes = "Preserve"
match_arm_blocks = false
match_block_trailing_comma = true
trailing_comma = "Vertical"
trailing_semicolon = false
```